### PR TITLE
support dynamic gas price

### PR DESCRIPTION
**Pull request by LeoGuo621**
Closes #<issue number>.

### Reviewers

Main reviewers:

- @John
- @Doe

Codeowner reviewers:

- @-Alice
- @-Bob

### CODE REVIEW

File: jsonrpc/endpoints_eth.go [hunk 1]
```diff
@@ -221,7 +226,30 @@ func (e *EthEndpoints) GasPrice() (interface{}, types.Error) {
...
+		e.cfg.DynamicGP = getApolloConfig().DynamicGP
+		getApolloConfig().RUnlock()
+	}
+	if e.cfg.DynamicGP.Enabled {
+		log.Debug("enable dynamic gas price")
+		// judge if there is congestion
+		isCongested, err := e.isCongested(ctx)
+		if err != nil {
+			log.Errorf("failed to count pool txs by status pending while judging if the pool is congested: ", err)
+			return "0x0", nil
```
Comment (giskook) [inline jsonrpc/endpoints_eth.go:241]: If there is some issue on the `isCongested` I think we'd better use the original gas price?

Comment (LeoGuo621) [reply jsonrpc/endpoints_eth.go:241]: I agree

**Review by giskook — APPROVED** [review]

File: jsonrpc/endpoints_eth.go [hunk 2]
```diff
@@ -221,7 +226,30 @@ func (e *EthEndpoints) GasPrice() (interface{}, types.Error) {
 	if err != nil {
 		return "0x0", nil
 	}
```
Comment (zjg555543) [inline jsonrpc/endpoints_eth.go:228]: Plz check rpc GasPrice's performance

**Review by zjg555543 — APPROVED** [review]

File: jsonrpc/endpoints_eth.go (hunk 2 above)
Comment (LeoGuo621) [reply jsonrpc/endpoints_eth.go:228]: Done. Not much difference in performance

**Pull request merged**